tests/unit/lib/SGSparseMatrix_unittest.cc

```diff
@@ -23,138 +25,203 @@ using namespace Eigen;
-
-	EXPECT_NEAR(r.norm(), 22.80350850198275836078, 1E-16);
+  const int32_t size=10;
+  const int32_t num_feat=size/2;
+
+  SGSparseMatrix<complex128_t> m(size, size);
+  for (index_t i=0; i<size; ++i)
+    {
+      m.sparse_matrix[i]=SGSparseVector<complex128_t>(num_feat);
+      for (index_t j=0; j<num_feat; ++j)
+        {
+          SGSparseVectorEntry<complex128_t> entry;
+          entry.feat_index=(j+1)*2;
+          entry.entry=complex128_t(0.5, 0.75);
+          m.sparse_matrix[i].features[j]=entry;
+        }
+    }
+
+  SGVector<int32_t> v(size);
+  v.set_const(2);
+
+  SGVector<complex128_t> result=m*v;
+  Map<VectorXcd> r(result.vector, result.vlen);
+
+  EXPECT_NEAR(r.norm(), 22.80350850198275836078, 1E-16);
 }
 
 TEST(SGSparseMatrix, multiply_complex128_float64)
 {
-	const int32_t size=10;
-	const int32_t num_feat=size/2;
-
-	SGSparseMatrix<complex128_t> m(size, size);
-	for (index_t i=0; i<size; ++i)
-	{
-		m.sparse_matrix[i]=SGSparseVector<complex128_t>(num_feat);
-		for (index_t j=0; j<num_feat; ++j)
-		{
-			SGSparseVectorEntry<complex128_t> entry;
-			entry.feat_index=(j+1)*2;
-			entry.entry=complex128_t(0.5, 0.75);
-			m.sparse_matrix[i].features[j]=entry;
-		}
-	}
-
-	SGVector<float64_t> v(size);
-	v.set_const(2);
-
-	SGVector<complex128_t> result=m*v;
-	Map<VectorXcd> r(result.vector, result.vlen);
-
-	EXPECT_NEAR(r.norm(), 22.80350850198275836078, 1E-16);
+  const int32_t size=10;
+  const int32_t num_feat=size/2;
+
+  SGSparseMatrix<complex128_t> m(size, size);
+  for (index_t i=0; i<size; ++i)
+    {
+      m.sparse_matrix[i]=SGSparseVector<complex128_t>(num_feat);
+      for (index_t j=0; j<num_feat; ++j)
+        {
+          SGSparseVectorEntry<complex128_t> entry;
+          entry.feat_index=(j+1)*2;
+          entry.entry=complex128_t(0.5, 0.75);
+          m.sparse_matrix[i].features[j]=entry;
+        }
+    }
+
+  SGVector<float64_t> v(size);
+  v.set_const(2);
+
+  SGVector<complex128_t> result=m*v;
+  Map<VectorXcd> r(result.vector, result.vlen);
+
+  EXPECT_NEAR(r.norm(), 22.80350850198275836078, 1E-16);
 }
 #endif // HAVE_EIGEN3
 
 TEST(SGSparseMatrix, access_by_index)
 {
-	const index_t size=2;
+  CRandom randGenerator(0);
+  float64_t sparseLevel=0.1;
+  const index_t numberOfRows=50;
+  const index_t numberOfCols=100;
+
+  SGSparseMatrix<float64_t> matrix(numberOfRows, numberOfCols);
+
+  for (index_t rowIndex=0; rowIndex<numberOfRows; ++rowIndex)
+    for (index_t colIndex=0; colIndex<numberOfRows; ++colIndex){
+      float64_t randomNumber=randGenerator.random(0.0,1.0);
+      if (randomNumber<=sparseLevel){
+        matrix(colIndex,rowIndex)=randomNumber;
+      }
+      if (randomNumber<=sparseLevel)
+        EXPECT_EQ(matrix(colIndex,rowIndex), randomNumber);
+      else
+        EXPECT_EQ(matrix(colIndex,rowIndex), 0.0);
+    }
+}
 
-	SGSparseMatrix<int32_t> m(size, size);
-	for (index_t i=0; i<size; ++i)
-		m(i,i)=i+1;
-	m.sort_features();
+TEST(SGSparseMatrix, io_libsvm)
```

Comment:
Apart from the indentation of the braces, this test has not been changed, right? The new changes are very difficult to parse like this.